### PR TITLE
ARTEMIS-2358 - Add user and pass to database store to allow encryption

### PR DESCRIPTION
**Pull request by andytaylor**
https://issues.apache.org/jira/browse/ARTEMIS-2358

### CODE REVIEW

File: docs/user-manual/en/security.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ certificate this name is the name to which their certificate's DN maps. If
 server will simply use whatever user name (if any) the client provides. This
 option is `false` by default.
 
+## Rejecting messages without a Validated user
```
Comment (franz1981) [inline docs/user-manual/en/security.md]: This doesn't seem related to this PR

File: examples/features/sub-modules/inter-broker-bridge/artemis-jms-bridge/readme.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ This approach might be preferred if for example you only have access to the 5.x
 ```xml
    <bean id="5xConnectionFactory" class="org.apache.activemq.ActiveMQConnectionFactory">
       <property name="brokerURL" value="tcp://localhost:61616"/>
-      <property name="userName" value="admin"/>
+      <property name="user" value="admin"/>
```
Comment (franz1981) [inline examples/features/sub-modules/inter-broker-bridge/artemis-jms-bridge/readme.md]: This doesn't seem related to this PR (as above)

File: tests/activemq5-unit-tests/src/test/java/org/apache/activemq/usecases/receiver-secured.xml [hunk 3]
```diff
@@ -25,7 +25,7 @@
   <broker brokerName="receiver" persistent="false" useJmx="true" allowTempAutoCreationOnSend="true" xmlns="http://activemq.apache.org/schema/core">
 
     <networkConnectors>
-      <networkConnector uri="static:(tcp://localhost:62001)" userName="system" password="manager"/>
+      <networkConnector uri="static:(tcp://localhost:62001)" user="system" password="manager"/>
```
Comment (franz1981) [inline tests/activemq5-unit-tests/src/test/java/org/apache/activemq/usecases/receiver-secured.xml]: This doesn't seem related to this PR

File: tests/activemq5-unit-tests/src/test/java/org/apache/activemq/usecases/sender-secured.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
   <broker brokerName="sender" persistent="false" useJmx="true" xmlns="http://activemq.apache.org/schema/core">
 
     <networkConnectors>
-      <networkConnector uri="static:(tcp://localhost:62002)" userName="system" password="manager">
+      <networkConnector uri="static:(tcp://localhost:62002)" user="system" password="manager">
```
Comment (franz1981) [inline tests/activemq5-unit-tests/src/test/java/org/apache/activemq/usecases/sender-secured.xml]: This doesn't seem related to this PR

**Comment by andytaylor** [comment]
Ive merged some changes from @franz1981 to deal with comments
